Common/Data/SubscriptionManager.cs

```diff
@@ -268,17 +275,25 @@ public void SetDataManager(IAlgorithmSubscriptionManager subscriptionManager)
         /// </summary>
         /// <param name="subscription">The subscription configuration</param>
         /// <param name="consolidator">The consolidator</param>
+        /// <param name="desiredTickType">The desired tick type for the subscription. If not given is null.</param>
         /// <returns>true if the subscription is valid for the consolidator</returns>
-        public static bool IsSubscriptionValidForConsolidator(SubscriptionDataConfig subscription, IDataConsolidator consolidator)
+        public static bool IsSubscriptionValidForConsolidator(SubscriptionDataConfig subscription, IDataConsolidator consolidator, TickType? desiredTickType = null)
         {
             if (subscription.Type == typeof(Tick) &&
                 LeanData.IsCommonLeanDataType(consolidator.OutputType))
             {
-                var tickType = LeanData.GetCommonTickTypeForCommonDataTypes(
+                if (desiredTickType == null)
+                {
+                    var tickType = LeanData.GetCommonTickTypeForCommonDataTypes(
                     consolidator.OutputType,
                     subscription.Symbol.SecurityType);
 
-                return subscription.TickType == tickType;
+                    return subscription.TickType == tickType;
+                }
+                else if (subscription.TickType != desiredTickType)
```

Comment:
If subscription's tick type is the same as the desired tick type, we coudl return true, we don't really need the method to continue doing any more checks outside of this `if`.
Maybe do this in a single check? Like doing `var tickType = desiredTickType ?? LeanData.GetCommonTickTypeForCommonDataTypes(consolidator.OutputType, subscription.Symbol.SecurityType);` and then just `return subscription.TickType == tickType;`?

Reply:
Hmm I see, that could be an option. However, I still wonder why don't you take into consideration this last check `consolidator.InputType.IsAssignableFrom(subscription.Type)`